Weather App/view/SearchLocationViewController.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  SearchLocationViewController.swift
+//  Weather App
+//
+//  Created by Shikhar Sharma on 23/03/22.
+//
+
+import UIKit
+
+class SearchLocationViewController: UIViewController {
+    
+    @IBOutlet private weak var locationSearchBar: UISearchBar!
+    @IBOutlet private weak var locationSuggestionTable: UITableView!
+    
+    static let id = "searchLocationViewController"
+    
+    public let searchLocVM = SearchLocationViewModel()
+    public var currentWeatherViewModel = WeatherViewModel()
+    public var searchSuggestions : [SearchSuggestion] = []
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        searchLocVM.Delegate = self
+        locationSearchBar.delegate = self
+        locationSearchBar.placeholder = "Enter Location Name"
+        locationSuggestionTable.delegate = self
+        locationSuggestionTable.dataSource = self
+        
+        // data binding
+        searchLocVM.searchSuggestion.bind { [weak self] data in
+            self?.searchSuggestions = data
+            self?.locationSuggestionTable.reloadData()
+        }
+    }
+
+}
+
+extension SearchLocationViewController : UISearchBarDelegate {
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        if (searchText.count > 3) {
+            searchLocVM.getSuggestions(query: searchText)
+        }
+    }
+}
+
+extension SearchLocationViewController : UITableViewDelegate, UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        self.searchSuggestions.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let item = self.searchSuggestions[indexPath.row]
+        let cell = tableView.dequeueReusableCell(withIdentifier: LocationSuggestionTableViewCell.id) as! LocationSuggestionTableViewCell
+        cell.nameLabel.text = item.name
```

Comment:
setting display elements is preferable to be done on willDisplay

Reply:
not getting how to implement this.

Reply:
Done